pkg/cloud/baremetal/actuators/machine/actuator_test.go

```diff
@@ -1630,3 +1635,103 @@ func TestDeleteOfBareMetalHostDeletesMachine(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdateUserData(t *testing.T) {
+	scheme := runtime.NewScheme()
+	clusterapis.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+
+	// Setup a fake https endpoint to server full ignition
+	server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "Full Ignition")
+	}))
+	defer server.Close()
+
+	cert := server.Certificate()
+	certInPem := pem.EncodeToMemory(
+		&pem.Block{
+			Type:  "CERTIFICATE",
+			Bytes: cert.Raw,
+		},
+	)
+	certB64 := base64.URLEncoding.EncodeToString(certInPem)
+
+	testCases := []struct {
+		Scenario     string
+		IgnitionData string
+		MachineName  string
+		ExpectUpdate bool
+	}{
+		{
+			Scenario: "user data has both url and ca cert",
+			IgnitionData: "{\"ignition\":{\"config\":{\"append\":[{\"source\":\"" +
+				server.URL +
+				"\",\"verification\":{}}]},\"security\":{\"tls\":{\"certificateAuthorities\":[{\"source\":\"data:text/plain;charset=utf-8;base64," +
+				certB64 +
+				"\",\"verification\":{}}]}},\"timeouts\":{},\"version\":\"2.2.0\"},\"networkd\":{},\"passwd\":{},\"storage\":{},\"systemd\":{}}",
+			ExpectUpdate: true,
+		},
+		{
+			Scenario: "user data has url but no ca cert",
+			IgnitionData: "{\"ignition\":{\"config\":{\"append\":[{\"source\":\"" +
+				server.URL +
+				"\",\"verification\":{}}]},\"security\":{\"tls\":{\"certificateAuthorities\":[{\"source\":\"data:text/plain;charset=utf-8;base64," +
+				"" +
+				"\",\"verification\":{}}]}},\"timeouts\":{},\"version\":\"2.2.0\"},\"networkd\":{},\"passwd\":{},\"storage\":{},\"systemd\":{}}",
+			ExpectUpdate: true,
+		},
+		{
+			Scenario: "user data has no url",
+			IgnitionData: "{\"ignition\":{\"config\":{\"append\":[{\"source\":\"" +
```

Comment:
Golang tip, you can use backticks for verbatim data and avoid all this escaping

Here's an example from gohpercloud's tests:
https://github.com/gophercloud/gophercloud/blob/master/openstack/baremetal/v1/allocations/testing/fixtures.go#L14-L67

Reply:
Awesome. This is indeed simpler. Changed in the new commit. Thanks for the tip.